articles/rendering-html-pages-as-a-http-server-response-using-node-js/index.md

```diff
@@ -0,0 +1,333 @@
+
+### Prerequisites
+
+Basic knowledge of using [Node.js](https://www.youtube.com/watch?v=fBNz5xF-Kx4) and [Express.js](/engineering-education/express/)
+
+### What we will cover
+
+- [Prerequisites](#prerequisites)
+- [What we will cover](#what-we-will-cover)
+- [Setting up](#setting-up)
+- [Rendering inline Html elements as HTTP responses](#rendering-inline-html-elements-as-http-responses)
+- [Render HTML web pages as server responses](#render-html-web-pages-as-server-responses)
+- [Parsering form data to the server using the HTML forms](#parsering-form-data-to-the-server-using-the-html-forms)
+- [Conclusion](#conclusion)
+
+### Setting up
+
+- [Download Node.js](https://nodejs.org/en/) and install it. Run `node -v` to test if the installation was successful.
+
+```bash
+$ node -v
+v12.18.3 ## installed node.js version
+```
+
+- Create a Node.js project/directory. Initialize the project within the directory you have created. Use `npm init -y` to automatically initialize this Node.js project or check this [guide](/engineering-education/beginner-guide-to-npm/) to learn more about using NPM.
+- Install the Express framework using `npm install express`. Check this [guide](/engineering-education/express/) to learn more about of Express.
+- We will create a server using Express. You may want to hook the server with Nodemon. It is an optional package (installed globally) which automatically restarts the server after saving server-side code changes. Go ahead and install Nodemon using `npm install -g nodemon`. Check this [guide](https://www.npmjs.com/package/nodemon) to learn more about Nodemon.
+
+### Rendering inline Html elements as HTTP responses
+
+Below is a [simple hello world HTTP server](https://expressjs.com/en/starter/hello-world.html) (`app.js`) listening on port 3000.
+
+```js
+const express = require("express");
+const app = express();
+
+app.listen(3000, () => {
+  console.log("Application started and Listening on port 3000");
+});
+
+app.get("/", (req, res) => {
+  res.send("Hello world!");
+});
+```
+
+>Run `nodemon` to start the server.
+
+Whenever the server is running, accessing the route `http://localhost:3000/`, will give you plain text `hello world!`.
+
+We can use the same server to render HTML elements as the server response instead of sending plain text.
+
+Here is a list of some HTML elements. We can render them directly into our server by specifying the response to send when the default route is accessed.
+
+```js
+const express = require("express");
+const app = express();
+
+app.listen(3000, () => {
+  console.log("Application started and Listening on port 3000");
+});
+
+app.get("/", (req, res) => {
+  res.send("<html> <head>server Response</head><body><h1> This page was render direcly from the server <p>Hello there welcome to my website</p></h1></body></html>");
+});
+```
+
+Restart the server and open route `http://localhost:3000/` on the browser.
+
+![](server-rendered-html-elements.jpg)
```

Comment:
Make sure to add text related to the content of the image. Also, modify all the image URLs to the following.
```![alt text](/engineering-education//rendering-html-pages-as-a-http-server-response-using-node-js/server-rendered-html-elements.jpg)```